specs/governance/gov-token.md

```diff
@@ -24,72 +24,82 @@
-to mint an arbitrary number of new tokens to an specific address. This function MUST only be called by the contract
+to mint an arbitrary number of new tokens to a specific address. This function MUST only be called by the contract
 owner, the `MintManager`, as enforced by the `onlyOwner` modifier inherited from the `Ownable` contract. When tokens
-are minted, the voting power of the recipient address MUST be updated accordingly. The total token supply is capped to
-`2^208^ - 1` to prevent overflow risks in the voting system. If the total supply exceeds this limit,
-`_mint(address,uint256)`, as inherited from `ERC20Votes`, MUST revert.
+are minted, the voting power of the recipient address MUST be updated accordingly in the `Alligator` contract via the
+transfer hook. The total token supply is capped to `2^208^ - 1` to prevent overflow risks in the voting system. If the
+total supply exceeds this limit, `_mint(address,uint256)`, as inherited from `ERC20Votes`, MUST revert.
 
 ### Token Burning
 
 The contract MUST allow token holders to burn their own tokens using the inherited `burn(uint256)` or
 `burnFrom(address,uint256)` functions inherited from `ERC20Burnable`. When tokens are burned, the total supply and the
-holder's voting power MUST be reduced accordingly.
+holder's voting power MUST be reduced accordingly in the `Alligator` contract via the transfer hook.
 
 ### Voting Power
 
 Each token corresponds to one unit of voting power.
 By default, token balance does not account for voting power. To have their voting power counted, token holders MUST delegate
 their voting power to an address (can be their own address).
-The contract MUST offer public accessors for quering voting power, as outlined below.
+The contract MUST offer public accessors for querying voting power, as outlined below.
 
 #### Queries
 
-- The `getVotes()(uint256)` function MUST return the current voting power of an address.
+- The `getVotes()(uint256)` function MUST retrieve the current voting power of an address from the `Alligator` contract.
 - The `getPastVotes(address,uint256)(uint256)` function MUST allow querying the voting power of an address at a specific
-  block number in the past.
+  block number in the past from the `Alligator` contract.
 - The `getPastTotalSupply(uint256)(uint256)` function MUST return the total voting power at a specific block number in
-  the past.
+  the past from the `Alligator` contract.
 
 ### Delegation
 
 Vote power can be delegated either by calling the `delegate(address)` function directly (to delegate as the `msg.sender`)
 or by providing a signature to be used with function `delegateBySig(address,uint256,uint256,uint8,bytes32,bytes32)`,
-as inherited from `ERC20Votes`.
+as inherited from `ERC20Votes`. These functions are modified to forward the calls to the `Alligator` contract using an
+advanced delegation rule that mimics basic delegation.
```

Comment:
Above we also said "using an advanced delegation rule that mimics basic delegation", but that feels vague to me so I've been deleting it in my suggestions. Feel free to rephrase as appropriate!
```suggestion
as inherited from `ERC20Votes`. These functions are modified to forward the calls to the `Alligator` contract which implements the required logic.
```